python/interpret-core/interpret/glassbox/_ebm/_ebm.py

```diff
@@ -551,6 +541,18 @@ def fit(self, X, y, sample_weight=None, bags=None, init_score=None):  # noqa: C9
 
         # TODO: check the other inputs for common mistakes here
 
+        if (_y := np.array(y)).ndim == 2 and _y.shape[1] == 1:
+            # TODO: clean_dimensions will remove the dimensions if size 1.
+            # Probably this should be moved. Is hot-fix for sklearn compliance.
+            warn(
+                (
+                    "A column-vector y was passed when a 1d array was"
+                    " expected. Please change the shape of y to "
+                    "(n_samples,), for example using ravel()."
+                ),
+                DataConversionWarning,
+                stacklevel=0,
+            )
```

Comment:
This part should most likely be moved to `remove_extra_dimensions`. However, I assume that `clean_dimensions` is not only used for `y`, as you explicitly pass the name to the functions, so it's probably wrong to have the generic check in there.

Reply:
Hi @DerWeh -- We actually accept these bad inputs (perhaps controversial) provided we can be reasonably confident that we understand the original intent of the caller.  So, for example if y had the shape (1, 76, 1) and X had the shape (76, 100), that works for us since the caller's intent is clear.  For y we can basically drop any dimension that's of length 1, and if all dimensions are length 1, then y has one sample and 1 feature.  For X, we know how many samples there by processing y first, so we ensure that the number of samples matches up, and we can again drop any extra dimensions that are of length 1 leaving the last dimension that isn't length 1 as the # of features.  I can see why scikit-learn doesn't want all their components to handle these kinds of weird/bad inputs because they'd want agreement between all their components which were written by different people, but we don't have that same issue.

The main place this comes in handy is when someone slices off a column but forgets to remove the extra dimension leading to y having shape=(N, 1)

Reply:
Sure, scikit-learn is a bit more pedantic and issues warnings (which could be filtered). But I agree, it hardly should be an issue. I'll drop the `DataConversionWarning`.